parquet/tests/simple_variant_integration.rs

```
@@ -0,0 +1,1253 @@
+                    test_case.case_number,
+                    test_case.test.as_deref().unwrap_or("unknown test")
+                );
+                ignored += 1;
+                continue;
+            }
+
+            match self.run_single_test(test_case) {
+                Ok(()) => {
+                    println!(
+                        "PASSED: case-{:03} - {}",
+                        test_case.case_number,
+                        test_case.test.as_deref().unwrap_or("unknown test")
+                    );
+                    passed += 1;
+                }
+                Err(e) => {
+                    println!(
+                        "FAILED: case-{:03} - {} - Error: {}",
+                        test_case.case_number,
+                        test_case.test.as_deref().unwrap_or("unknown test"),
+                        e
+                    );
+                    failed += 1;
+                }
+            }
+        }
+
+        println!("\nTest Results:");
+        println!("  Passed: {}", passed);
+        println!("  Failed: {}", failed);
+        println!("  Ignored: {}", ignored);
+        println!("  Total: {}", passed + failed + ignored);
+
+        if failed > 0 {
+            Err(format!("{} tests failed", failed).into())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Run a single test case
+    fn run_single_test(&self, test_case: &VariantTestCase) -> Result<(), Box<dyn Error>> {
+        match &test_case.test_category {
+            TestCategory::ErrorHandling => {
+                // For error cases, we expect the parsing/validation to fail
+                self.run_error_test(test_case)
+            }
+            _ => {
+                // For normal cases, run standard validation
+                self.run_success_test(test_case)
+            }
+        }
+    }
+
+    /// Run a test case that should succeed
+    fn run_success_test(&self, test_case: &VariantTestCase) -> Result<(), Box<dyn Error>> {
```

In tests, I personally find it helpful to fail fast if there is a problem

Thus one pattern you might consider here returning `()` and using `expect` or `unwrap` in the test code 

So something instead of
```rust
        let struct_arrays = self.read_parquet_file(test_case)?;
```

More like 

```rust
        let struct_arrays = self.read_parquet_file(test_case).expect("error reading parquet file");
```